utils/src/sequence/prefixed_u64.rs

```diff
@@ -24,12 +24,11 @@ pub enum Error {
 pub struct U64([u8; u64::SIZE + 1]);
 
 impl U64 {
-    pub fn new(prefix: u8, value: u64) -> Self {
-        let mut arr = [0; u64::SIZE + 1];
-        arr[0] = prefix;
-        arr[1..].copy_from_slice(&u64::to_be_bytes(value));
-
-        Self(arr)
+    pub const fn new(prefix: u8, value: u64) -> Self {
+        // TODO: #![feature(const_index)]
+        // https://github.com/rust-lang/rust/issues/143775
+        let [b0, b1, b2, b3, b4, b5, b6, b7] = value.to_be_bytes();
+        Self([prefix, b0, b1, b2, b3, b4, b5, b6, b7])
     }
```

Comment:
For readability, somewhat inclined to just drop into unsafe and do a direct memory copy.

```suggestion
    pub const fn new(prefix: u8, value: u64) -> Self {
        let mut arr = [0; u8::SIZE + u64::SIZE];
        arr[0] = prefix;
        
        // SAFETY: `arr` is known to be 9 bytes long and valid.
        unsafe {
            core::ptr::copy_nonoverlapping(
                u64::to_be_bytes(value).as_ptr(),
                arr.as_mut_ptr().add(u8::SIZE),
                u64::SIZE
            );
        }
        
        Self(arr)
    }
```